pkg/controller/controller.go

```diff
@@ -143,10 +143,11 @@ func (c *ControllerServer) CreateVolume(ctx context.Context, req *csi.CreateVolu
 		copiedDrive.Status.FreeCapacity = copiedDrive.Status.FreeCapacity - req.GetCapacityRange().GetRequiredBytes()
 		copiedDrive.Status.AllocatedCapacity = copiedDrive.Status.AllocatedCapacity + req.GetCapacityRange().GetRequiredBytes()
 		copiedDrive.Status.DriveStatus = direct_csi.Online
+		copiedDrive.ObjectMeta.Finalizers = utils.AddFinalizer(copiedDrive.ObjectMeta.Finalizers, vol.ObjectMeta.Name)
```

Comment:
I think here you don't follow the [naming convention for finalizers](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#finalizers):

> Identifiers of custom finalizers consist of a domain name, a forward slash and the name of the finalizer.

Reply:
```suggestion
		copiedDrive.ObjectMeta.Finalizers = utils.AddFinalizer(copiedDrive.ObjectMeta.Finalizers, fmt.Sprintf("%s/%s, v1alpha1.SchemeGroupVersion.Group, vol.ObjectMeta.Name))
```

Reply:
This is a good catch. Please look into this: https://github.com/kubernetes/apimachinery/blob/master/pkg/api/validation/objectmeta.go#L103